src/Containers/Microsoft.NET.Build.Containers/LocalDaemons/DockerCli.cs

```diff
@@ -56,32 +78,43 @@ public async Task LoadAsync(BuiltImage image, ImageReference sourceReference, Im
         }
     }
 
-    public Task<bool> IsAvailableAsync(CancellationToken cancellationToken)
+    public async Task<bool> IsAvailableAsync(CancellationToken cancellationToken)
     {
-        return Task.FromResult(IsAvailable());
-    }
+        string? commandPath = await GetCommandPathAsync(cancellationToken);
+        if (commandPath is null)
+        {
+            logger($"Cannot find {DockerCommand}/{PodmanCommand} executable.");
+            return false;
+        }
 
-    ///<inheritdoc/>
-    public bool IsAvailable()
-    {
         try
         {
-            JsonDocument config = GetConfig();
-
-            if (!config.RootElement.TryGetProperty("ServerErrors", out JsonElement errorProperty))
-            {
-                return true;
-            }
-            else if (errorProperty.ValueKind == JsonValueKind.Array && errorProperty.GetArrayLength() == 0)
-            {
-                return true;
-            }
-            else
+            switch (commandPath)
             {
-                // we have errors, turn them into a string and log them
-                string messages = string.Join(Environment.NewLine, errorProperty.EnumerateArray());
-                logger($"The daemon server reported errors: {messages}");
-                return false;
+                case DockerCommand:
+                {
+                    JsonDocument config = GetConfig();
+
+                    if (!config.RootElement.TryGetProperty("ServerErrors", out JsonElement errorProperty))
+                    {
+                        return true;
+                    }
+                    else if (errorProperty.ValueKind == JsonValueKind.Array && errorProperty.GetArrayLength() == 0)
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        // we have errors, turn them into a string and log them
+                        string messages = string.Join(Environment.NewLine, errorProperty.EnumerateArray());
+                        logger($"The daemon server reported errors: {messages}");
+                        return false;
+                    }
+                }
+                case PodmanCommand:
+                    return true;
```

Comment:
I will add a "podman version" invocation here to ensure the tool is installed when this method returns true.